### ISSUE TITLE
DSR(direct routing) with RKE2 and Kube Router on Public VPS's not working

### ISSUE THREAD

**Issue by hox** [cilium/cilium#22951]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Currently I have a two-node cluster on public VPS’s and am trying to setup native routing with DSR. There is an example nginx server running on a NodePort inside of the cluster.

The issue I am running into is, once Cilium is setup with DSR & kube-router, the nginx server is only accessible through the public IP of the node it is actually running on. Attached are the logs and status/health commands for Cilium as well as the configuration for kube-router.

### Cilium Version

1.12.5

### Kernel Version

Linux node-adm-01 5.10.0-11-amd64 #1 SMP Debian 5.10.92-1 (2022-01-18) x86_64 GNU/Linux
Linux node-adm-02 5.10.0-11-amd64 #1 SMP Debian 5.10.92-1 (2022-01-18) x86_64 GNU/Linux

### Kubernetes Version

v1.24.9

### Relevant log output

```shell
level=info msg="Started gops server" address="127.0.0.1:9890" subsys=daemon
level=info msg="Memory available for map entries (0.003% of 4122218496B): 10305546B" subsys=config
level=info msg="option bpf-ct-global-tcp-max set by dynamic sizing to 131072" subsys=config
level=info msg="option bpf-ct-global-any-max set by dynamic sizing to 65536" subsys=config
level=info msg="option bpf-nat-global-max set by dynamic sizing to 131072" subsys=config
level=info msg="option bpf-neigh-global-max set by dynamic sizing to 131072" subsys=config
level=info msg="option bpf-sock-rev-map-max set by dynamic sizing to 65536" subsys=config
level=info msg="  --agent-health-port='9879'" subsys=daemon
level=info msg="  --agent-labels=''" subsys=daemon
level=info msg="  --agent-not-ready-taint-key='node.cilium.io/agent-not-ready'" subsys=daemon
level=info msg="  --allocator-list-timeout='3m0s'" subsys=daemon
level=info msg="  --allow-icmp-frag-needed='true'" subsys=daemon
level=info msg="  --allow-localhost='auto'" subsys=daemon
level=info msg="  --annotate-k8s-node='false'" subsys=daemon
level=info msg="  --api-rate-limit=''" subsys=daemon
level=info msg="  --arping-refresh-period='30s'" subsys=daemon
level=info msg="  --auto-create-cilium-node-resource='true'" subsys=daemon
level=info msg="  --auto-direct-node-routes='false'" subsys=daemon
level=info msg="  --bgp-announce-lb-ip='false'" subsys=daemon
level=info msg="  --bgp-announce-pod-cidr='false'" subsys=daemon
level=info msg="  --bgp-config-path='/var/lib/cilium/bgp/config.yaml'" subsys=daemon
level=info msg="  --bpf-ct-global-any-max='262144'" subsys=daemon
level=info msg="  --bpf-ct-global-tcp-max='524288'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-any='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp='6h0m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp-fin='10s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp-syn='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-any='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-tcp='6h0m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-tcp-grace='1m0s'" subsys=daemon
level=info msg="  --bpf-filter-priority='1'" subsys=daemon
level=info msg="  --bpf-fragments-map-max='8192'" subsys=daemon
level=info msg="  --bpf-lb-acceleration='disabled'" subsys=daemon
level=info msg="  --bpf-lb-affinity-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-algorithm='random'" subsys=daemon
level=info msg="  --bpf-lb-dev-ip-addr-inherit=''" subsys=daemon
level=info msg="  --bpf-lb-dsr-dispatch='opt'" subsys=daemon
level=info msg="  --bpf-lb-dsr-l4-xlate='frontend'" subsys=daemon
level=info msg="  --bpf-lb-external-clusterip='false'" subsys=daemon
level=info msg="  --bpf-lb-maglev-hash-seed='JLfvgnHc2kaSUFaI'" subsys=daemon
level=info msg="  --bpf-lb-maglev-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-maglev-table-size='16381'" subsys=daemon
level=info msg="  --bpf-lb-map-max='65536'" subsys=daemon
level=info msg="  --bpf-lb-mode='dsr'" subsys=daemon
level=info msg="  --bpf-lb-rev-nat-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-rss-ipv4-src-cidr=''" subsys=daemon
level=info msg="  --bpf-lb-rss-ipv6-src-cidr=''" subsys=daemon
level=info msg="  --bpf-lb-service-backend-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-service-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-sock='false'" subsys=daemon
level=info msg="  --bpf-lb-sock-hostns-only='false'" subsys=daemon
level=info msg="  --bpf-lb-source-range-map-max='0'" subsys=daemon
level=info msg="  --bpf-map-dynamic-size-ratio='0.0025'" subsys=daemon
level=info msg="  --bpf-nat-global-max='524288'" subsys=daemon
level=info msg="  --bpf-neigh-global-max='524288'" subsys=daemon
level=info msg="  --bpf-policy-map-max='16384'" subsys=daemon
level=info msg="  --bpf-root='/sys/fs/bpf'" subsys=daemon
level=info msg="  --bpf-sock-rev-map-max='262144'" subsys=daemon
level=info msg="  --bypass-ip-availability-upon-restore='false'" subsys=daemon
level=info msg="  --certificates-directory='/var/run/cilium/certs'" subsys=daemon
level=info msg="  --cflags=''" subsys=daemon
level=info msg="  --cgroup-root='/run/cilium/cgroupv2'" subsys=daemon
level=info msg="  --cluster-health-port='4240'" subsys=daemon
level=info msg="  --cluster-id='0'" subsys=daemon
level=info msg="  --cluster-name='default'" subsys=daemon
level=info msg="  --clustermesh-config='/var/lib/cilium/clustermesh/'" subsys=daemon
level=info msg="  --cmdref=''" subsys=daemon
level=info msg="  --cni-chaining-mode=''" subsys=daemon
level=info msg="  --config=''" subsys=daemon
level=info msg="  --config-dir='/tmp/cilium/config-map'" subsys=daemon
level=info msg="  --conntrack-gc-interval='0s'" subsys=daemon
level=info msg="  --crd-wait-timeout='5m0s'" subsys=daemon
level=info msg="  --datapath-mode='veth'" subsys=daemon
level=info msg="  --debug='false'" subsys=daemon
level=info msg="  --debug-verbose=''" subsys=daemon
level=info msg="  --derive-masquerade-ip-addr-from-device=''" subsys=daemon
level=info msg="  --devices=''" subsys=daemon
level=info msg="  --direct-routing-device=''" subsys=daemon
level=info msg="  --disable-cnp-status-updates='true'" subsys=daemon
level=info msg="  --disable-conntrack='false'" subsys=daemon
level=info msg="  --disable-endpoint-crd='false'" subsys=daemon
level=info msg="  --disable-envoy-version-check='false'" subsys=daemon
level=info msg="  --disable-iptables-feeder-rules=''" subsys=daemon
level=info msg="  --dns-max-ips-per-restored-rule='1000'" subsys=daemon
level=info msg="  --dns-policy-unload-on-shutdown='false'" subsys=daemon
level=info msg="  --dnsproxy-concurrency-limit='0'" subsys=daemon
level=info msg="  --dnsproxy-concurrency-processing-grace-period='0s'" subsys=daemon
level=info msg="  --egress-masquerade-interfaces=''" subsys=daemon
level=info msg="  --egress-multi-home-ip-rule-compat='false'" subsys=daemon
level=info msg="  --enable-auto-protect-node-port-range='true'" subsys=daemon
level=info msg="  --enable-bandwidth-manager='false'" subsys=daemon
level=info msg="  --enable-bbr='false'" subsys=daemon
level=info msg="  --enable-bgp-control-plane='false'" subsys=daemon
level=info msg="  --enable-bpf-clock-probe='true'" subsys=daemon
level=info msg="  --enable-bpf-masquerade='true'" subsys=daemon
level=info msg="  --enable-bpf-tproxy='false'" subsys=daemon
level=info msg="  --enable-cilium-endpoint-slice='false'" subsys=daemon
level=info msg="  --enable-custom-calls='false'" subsys=daemon
level=info msg="  --enable-endpoint-health-checking='true'" subsys=daemon
level=info msg="  --enable-endpoint-routes='false'" subsys=daemon
level=info msg="  --enable-envoy-config='false'" subsys=daemon
level=info msg="  --enable-external-ips='true'" subsys=daemon
level=info msg="  --enable-health-check-nodeport='true'" subsys=daemon
level=info msg="  --enable-health-checking='true'" subsys=daemon
level=info msg="  --enable-host-firewall='false'" subsys=daemon
level=info msg="  --enable-host-legacy-routing='false'" subsys=daemon
level=info msg="  --enable-host-port='true'" subsys=daemon
level=info msg="  --enable-host-reachable-services='false'" subsys=daemon
level=info msg="  --enable-hubble='true'" subsys=daemon
level=info msg="  --enable-hubble-recorder-api='true'" subsys=daemon
level=info msg="  --enable-icmp-rules='true'" subsys=daemon
level=info msg="  --enable-identity-mark='true'" subsys=daemon
level=info msg="  --enable-ip-masq-agent='false'" subsys=daemon
level=info msg="  --enable-ipsec='false'" subsys=daemon
level=info msg="  --enable-ipv4='true'" subsys=daemon
level=info msg="  --enable-ipv4-egress-gateway='false'" subsys=daemon
level=info msg="  --enable-ipv4-fragment-tracking='true'" subsys=daemon
level=info msg="  --enable-ipv4-masquerade='true'" subsys=daemon
level=info msg="  --enable-ipv6='false'" subsys=daemon
level=info msg="  --enable-ipv6-masquerade='true'" subsys=daemon
level=info msg="  --enable-ipv6-ndp='false'" subsys=daemon
level=info msg="  --enable-k8s-api-discovery='false'" subsys=daemon
level=info msg="  --enable-k8s-endpoint-slice='true'" subsys=daemon
level=info msg="  --enable-k8s-event-handover='false'" subsys=daemon
level=info msg="  --enable-k8s-terminating-endpoint='true'" subsys=daemon
level=info msg="  --enable-l2-neigh-discovery='true'" subsys=daemon
level=info msg="  --enable-l7-proxy='true'" subsys=daemon
level=info msg="  --enable-local-node-route='true'" subsys=daemon
level=info msg="  --enable-local-redirect-policy='false'" subsys=daemon
level=info msg="  --enable-mke='false'" subsys=daemon
level=info msg="  --enable-monitor='true'" subsys=daemon
level=info msg="  --enable-node-port='false'" subsys=daemon
level=info msg="  --enable-policy='default'" subsys=daemon
level=info msg="  --enable-recorder='false'" subsys=daemon
level=info msg="  --enable-remote-node-identity='true'" subsys=daemon
level=info msg="  --enable-runtime-device-detection='false'" subsys=daemon
level=info msg="  --enable-selective-regeneration='true'" subsys=daemon
level=info msg="  --enable-service-topology='false'" subsys=daemon
level=info msg="  --enable-session-affinity='false'" subsys=daemon
level=info msg="  --enable-svc-source-range-check='true'" subsys=daemon
level=info msg="  --enable-tracing='false'" subsys=daemon
level=info msg="  --enable-unreachable-routes='false'" subsys=daemon
level=info msg="  --enable-vtep='false'" subsys=daemon
level=info msg="  --enable-well-known-identities='false'" subsys=daemon
level=info msg="  --enable-wireguard='false'" subsys=daemon
level=info msg="  --enable-wireguard-userspace-fallback='false'" subsys=daemon
level=info msg="  --enable-xdp-prefilter='false'" subsys=daemon
level=info msg="  --enable-xt-socket-fallback='true'" subsys=daemon
level=info msg="  --encrypt-interface=''" subsys=daemon
level=info msg="  --encrypt-node='false'" subsys=daemon
level=info msg="  --endpoint-gc-interval='5m0s'" subsys=daemon
level=info msg="  --endpoint-interface-name-prefix=''" subsys=daemon
level=info msg="  --endpoint-queue-size='25'" subsys=daemon
level=info msg="  --endpoint-status=''" subsys=daemon
level=info msg="  --envoy-config-timeout='2m0s'" subsys=daemon
level=info msg="  --envoy-log=''" subsys=daemon
level=info msg="  --exclude-local-address=''" subsys=daemon
level=info msg="  --fixed-identity-mapping=''" subsys=daemon
level=info msg="  --force-local-policy-eval-at-source='true'" subsys=daemon
level=info msg="  --fqdn-regex-compile-lru-size='1024'" subsys=daemon
level=info msg="  --gops-port='9890'" subsys=daemon
level=info msg="  --host-reachable-services-protos='tcp,udp'" subsys=daemon
level=info msg="  --http-403-msg=''" subsys=daemon
level=info msg="  --http-idle-timeout='0'" subsys=daemon
level=info msg="  --http-max-grpc-timeout='0'" subsys=daemon
level=info msg="  --http-normalize-path='true'" subsys=daemon
level=info msg="  --http-request-timeout='3600'" subsys=daemon
level=info msg="  --http-retry-count='3'" subsys=daemon
level=info msg="  --http-retry-timeout='0'" subsys=daemon
level=info msg="  --hubble-disable-tls='false'" subsys=daemon
level=info msg="  --hubble-event-buffer-capacity='4095'" subsys=daemon
level=info msg="  --hubble-event-queue-size='0'" subsys=daemon
level=info msg="  --hubble-export-file-compress='false'" subsys=daemon
level=info msg="  --hubble-export-file-max-backups='5'" subsys=daemon
level=info msg="  --hubble-export-file-max-size-mb='10'" subsys=daemon
level=info msg="  --hubble-export-file-path=''" subsys=daemon
level=info msg="  --hubble-listen-address=':4244'" subsys=daemon
level=info msg="  --hubble-metrics=''" subsys=daemon
level=info msg="  --hubble-metrics-server=''" subsys=daemon
level=info msg="  --hubble-recorder-sink-queue-size='1024'" subsys=daemon
level=info msg="  --hubble-recorder-storage-path='/var/run/cilium/pcaps'" subsys=daemon
level=info msg="  --hubble-socket-path='/var/run/cilium/hubble.sock'" subsys=daemon
level=info msg="  --hubble-tls-cert-file='/var/lib/cilium/tls/hubble/server.crt'" subsys=daemon
level=info msg="  --hubble-tls-client-ca-files='/var/lib/cilium/tls/hubble/client-ca.crt'" subsys=daemon
level=info msg="  --hubble-tls-key-file='/var/lib/cilium/tls/hubble/server.key'" subsys=daemon
level=info msg="  --identity-allocation-mode='crd'" subsys=daemon
level=info msg="  --identity-change-grace-period='5s'" subsys=daemon
level=info msg="  --identity-restore-grace-period='10m0s'" subsys=daemon
level=info msg="  --install-egress-gateway-routes='false'" subsys=daemon
level=info msg="  --install-iptables-rules='true'" subsys=daemon
level=info msg="  --install-no-conntrack-iptables-rules='false'" subsys=daemon
level=info msg="  --ip-allocation-timeout='2m0s'" subsys=daemon
level=info msg="  --ip-masq-agent-config-path='/etc/config/ip-masq-agent'" subsys=daemon
level=info msg="  --ipam='kubernetes'" subsys=daemon
level=info msg="  --ipsec-key-file=''" subsys=daemon
level=info msg="  --iptables-lock-timeout='5s'" subsys=daemon
level=info msg="  --iptables-random-fully='false'" subsys=daemon
level=info msg="  --ipv4-native-routing-cidr='10.42.0.0/23'" subsys=daemon
level=info msg="  --ipv4-node='auto'" subsys=daemon
level=info msg="  --ipv4-pod-subnets=''" subsys=daemon
level=info msg="  --ipv4-range='auto'" subsys=daemon
level=info msg="  --ipv4-service-loopback-address='169.254.42.1'" subsys=daemon
level=info msg="  --ipv4-service-range='auto'" subsys=daemon
level=info msg="  --ipv6-cluster-alloc-cidr='f00d::/64'" subsys=daemon
level=info msg="  --ipv6-mcast-device=''" subsys=daemon
level=info msg="  --ipv6-native-routing-cidr=''" subsys=daemon
level=info msg="  --ipv6-node='auto'" subsys=daemon
level=info msg="  --ipv6-pod-subnets=''" subsys=daemon
level=info msg="  --ipv6-range='auto'" subsys=daemon
level=info msg="  --ipv6-service-range='auto'" subsys=daemon
level=info msg="  --join-cluster='false'" subsys=daemon
level=info msg="  --k8s-api-server=''" subsys=daemon
level=info msg="  --k8s-heartbeat-timeout='30s'" subsys=daemon
level=info msg="  --k8s-kubeconfig-path=''" subsys=daemon
level=info msg="  --k8s-namespace='kube-system'" subsys=daemon
level=info msg="  --k8s-require-ipv4-pod-cidr='false'" subsys=daemon
level=info msg="  --k8s-require-ipv6-pod-cidr='false'" subsys=daemon
level=info msg="  --k8s-service-cache-size='128'" subsys=daemon
level=info msg="  --k8s-service-proxy-name=''" subsys=daemon
level=info msg="  --k8s-sync-timeout='3m0s'" subsys=daemon
level=info msg="  --k8s-watcher-endpoint-selector='metadata.name!=kube-scheduler,metadata.name!=kube-controller-manager,metadata.name!=etcd-operator,metadata.name!=gcp-controller-manager'" subsys=daemon
level=info msg="  --keep-config='false'" subsys=daemon
level=info msg="  --kube-proxy-replacement='strict'" subsys=daemon
level=info msg="  --kube-proxy-replacement-healthz-bind-address=''" subsys=daemon
level=info msg="  --kvstore=''" subsys=daemon
level=info msg="  --kvstore-connectivity-timeout='2m0s'" subsys=daemon
level=info msg="  --kvstore-lease-ttl='15m0s'" subsys=daemon
level=info msg="  --kvstore-max-consecutive-quorum-errors='2'" subsys=daemon
level=info msg="  --kvstore-opt=''" subsys=daemon
level=info msg="  --kvstore-periodic-sync='5m0s'" subsys=daemon
level=info msg="  --label-prefix-file=''" subsys=daemon
level=info msg="  --labels=''" subsys=daemon
level=info msg="  --lib-dir='/var/lib/cilium'" subsys=daemon
level=info msg="  --local-max-addr-scope='252'" subsys=daemon
level=info msg="  --local-router-ipv4=''" subsys=daemon
level=info msg="  --local-router-ipv6=''" subsys=daemon
level=info msg="  --log-driver=''" subsys=daemon
level=info msg="  --log-opt=''" subsys=daemon
level=info msg="  --log-system-load='false'" subsys=daemon
level=info msg="  --max-controller-interval='0'" subsys=daemon
level=info msg="  --metrics=''" subsys=daemon
level=info msg="  --mke-cgroup-mount=''" subsys=daemon
level=info msg="  --monitor-aggregation='medium'" subsys=daemon
level=info msg="  --monitor-aggregation-flags='all'" subsys=daemon
level=info msg="  --monitor-aggregation-interval='5s'" subsys=daemon
level=info msg="  --monitor-queue-size='0'" subsys=daemon
level=info msg="  --mtu='0'" subsys=daemon
level=info msg="  --node-port-acceleration='disabled'" subsys=daemon
level=info msg="  --node-port-algorithm='random'" subsys=daemon
level=info msg="  --node-port-bind-protection='true'" subsys=daemon
level=info msg="  --node-port-mode='snat'" subsys=daemon
level=info msg="  --node-port-range='30000,32767'" subsys=daemon
level=info msg="  --policy-audit-mode='false'" subsys=daemon
level=info msg="  --policy-queue-size='100'" subsys=daemon
level=info msg="  --policy-trigger-interval='1s'" subsys=daemon
level=info msg="  --pprof='false'" subsys=daemon
level=info msg="  --pprof-port='6060'" subsys=daemon
level=info msg="  --preallocate-bpf-maps='false'" subsys=daemon
level=info msg="  --prepend-iptables-chains='true'" subsys=daemon
level=info msg="  --procfs='/host/proc'" subsys=daemon
level=info msg="  --prometheus-serve-addr=':9962'" subsys=daemon
level=info msg="  --proxy-connect-timeout='1'" subsys=daemon
level=info msg="  --proxy-gid='1337'" subsys=daemon
level=info msg="  --proxy-max-connection-duration-seconds='0'" subsys=daemon
level=info msg="  --proxy-max-requests-per-connection='0'" subsys=daemon
level=info msg="  --proxy-prometheus-port='0'" subsys=daemon
level=info msg="  --read-cni-conf=''" subsys=daemon
level=info msg="  --restore='true'" subsys=daemon
level=info msg="  --route-metric='0'" subsys=daemon
level=info msg="  --sidecar-istio-proxy-image='cilium/istio_proxy'" subsys=daemon
level=info msg="  --single-cluster-route='false'" subsys=daemon
level=info msg="  --socket-path='/var/run/cilium/cilium.sock'" subsys=daemon
level=info msg="  --sockops-enable='false'" subsys=daemon
level=info msg="  --state-dir='/var/run/cilium'" subsys=daemon
level=info msg="  --tofqdns-dns-reject-response-code='refused'" subsys=daemon
level=info msg="  --tofqdns-enable-dns-compression='true'" subsys=daemon
level=info msg="  --tofqdns-endpoint-max-ip-per-hostname='50'" subsys=daemon
level=info msg="  --tofqdns-idle-connection-grace-period='0s'" subsys=daemon
level=info msg="  --tofqdns-max-deferred-connection-deletes='10000'" subsys=daemon
level=info msg="  --tofqdns-min-ttl='3600'" subsys=daemon
level=info msg="  --tofqdns-pre-cache=''" subsys=daemon
level=info msg="  --tofqdns-proxy-port='0'" subsys=daemon
level=info msg="  --tofqdns-proxy-response-max-delay='100ms'" subsys=daemon
level=info msg="  --trace-payloadlen='128'" subsys=daemon
level=info msg="  --tunnel='disabled'" subsys=daemon
level=info msg="  --tunnel-port='0'" subsys=daemon
level=info msg="  --version='false'" subsys=daemon
level=info msg="  --vlan-bpf-bypass=''" subsys=daemon
level=info msg="  --vtep-cidr=''" subsys=daemon
level=info msg="  --vtep-endpoint=''" subsys=daemon
level=info msg="  --vtep-mac=''" subsys=daemon
level=info msg="  --vtep-mask=''" subsys=daemon
level=info msg="  --write-cni-conf-when-ready=''" subsys=daemon
level=info msg="     _ _ _" subsys=daemon
level=info msg=" ___|_| |_|_ _ _____" subsys=daemon
level=info msg="|  _| | | | | |     |" subsys=daemon
level=info msg="|___|_|_|_|___|_|_|_|" subsys=daemon
level=info msg="Cilium 1.12.5 701acde 2022-12-15T16:03:30-08:00 go version go1.18.9 linux/amd64" subsys=daemon
level=info msg="cilium-envoy  version: 48cbb2f88edca93ba80d1135dbb067a827b8adc2/1.21.5/Distribution/RELEASE/BoringSSL" subsys=daemon
level=info msg="clang (10.0.0) and kernel (5.10.0) versions: OK!" subsys=linux-datapath
level=info msg="linking environment: OK!" subsys=linux-datapath
level=info msg="Detected mounted BPF filesystem at /sys/fs/bpf" subsys=bpf
level=info msg="Mounted cgroupv2 filesystem at /run/cilium/cgroupv2" subsys=cgroups
level=info msg="Parsing base label prefixes from default label list" subsys=labels-filter
level=info msg="Parsing additional label prefixes from user inputs: []" subsys=labels-filter
level=info msg="Final label prefixes to be used for identity evaluation:" subsys=labels-filter
level=info msg=" - reserved:.*" subsys=labels-filter
level=info msg=" - :io\\.kubernetes\\.pod\\.namespace" subsys=labels-filter
level=info msg=" - :io\\.cilium\\.k8s\\.namespace\\.labels" subsys=labels-filter
level=info msg=" - :app\\.kubernetes\\.io" subsys=labels-filter
level=info msg=" - !:io\\.kubernetes" subsys=labels-filter
level=info msg=" - !:kubernetes\\.io" subsys=labels-filter
level=info msg=" - !:.*beta\\.kubernetes\\.io" subsys=labels-filter
level=info msg=" - !:k8s\\.io" subsys=labels-filter
level=info msg=" - !:pod-template-generation" subsys=labels-filter
level=info msg=" - !:pod-template-hash" subsys=labels-filter
level=info msg=" - !:controller-revision-hash" subsys=labels-filter
level=info msg=" - !:annotation.*" subsys=labels-filter
level=info msg=" - !:etcd_node" subsys=labels-filter
level=info msg="Auto-disabling \"enable-bpf-clock-probe\" feature since KERNEL_HZ cannot be determined" error="Cannot probe CONFIG_HZ" subsys=daemon
level=info msg="Using autogenerated IPv4 allocation range" subsys=node v4Prefix=10.251.0.0/16
level=info msg="Initializing daemon" subsys=daemon
level=info msg="Establishing connection to apiserver" host="https://159.223.122.251:6443" subsys=k8s
level=info msg="Connected to apiserver" subsys=k8s
level=info msg="Trying to auto-enable \"enable-node-port\", \"enable-external-ips\", \"bpf-lb-sock\", \"enable-host-port\", \"enable-session-affinity\" features" subsys=daemon
level=info msg="Inheriting MTU from external network interface" device=eth0 ipAddr=159.223.122.251 mtu=1500 subsys=mtu
level=info msg="Envoy: Starting xDS gRPC server listening on /var/run/cilium/xds.sock" subsys=envoy-manager
level=info msg="Restored backends from maps" failedBackends=0 restoredBackends=5 subsys=service
level=info msg="Restored services from maps" failedServices=0 restoredServices=7 subsys=service
level=info msg="Reading old endpoints..." subsys=daemon
level=info msg="Reusing previous DNS proxy port: 41577" subsys=daemon
level=info msg="Waiting until all Cilium CRDs are available" subsys=k8s
level=info msg="All Cilium CRDs have been found and are available" subsys=k8s
level=info msg="Retrieved node information from kubernetes node" nodeName=node-adm-01 subsys=k8s
level=info msg="Received own node information from API server" ipAddr.ipv4=159.223.122.251 ipAddr.ipv6="<nil>" k8sNodeIP=159.223.122.251 labels="map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux egress.rke2.io/cluster:true kubernetes.io/arch:amd64 kubernetes.io/hostname:node-adm-01 kubernetes.io/os:linux node-role.kubernetes.io/control-plane:true node-role.kubernetes.io/etcd:true node-role.kubernetes.io/master:true]" nodeName=node-adm-01 subsys=k8s v4Prefix=10.42.0.0/24 v6Prefix="<nil>"
level=info msg="k8s mode: Allowing localhost to reach local endpoints" subsys=daemon
level=info msg="Direct routing device detected" direct-routing-device=eth0 subsys=linux-datapath
level=info msg="Detected devices" devices="[eth0 eth1 tun-10424822832]" subsys=linux-datapath
level=info msg="Masquerading IP selected for device" device=eth0 ipv4=159.223.122.251 subsys=node
level=info msg="Masquerading IP selected for device" device=eth1 ipv4=10.116.0.4 subsys=node
level=info msg="Masquerading IP selected for device" device=tun-10424822832 ipv4=159.223.122.251 subsys=node
level=info msg="Enabling k8s event listener" subsys=k8s-watcher
level=info msg="Removing stale endpoint interfaces" subsys=daemon
level=info msg="Skipping kvstore configuration" subsys=daemon
level=info msg="Restored router address from node_config" file=/var/run/cilium/state/globals/node_config.h ipv4=10.42.0.87 ipv6="<nil>" subsys=node
level=info msg="Initializing node addressing" subsys=daemon
level=info msg="Initializing kubernetes IPAM" subsys=ipam v4Prefix=10.42.0.0/24 v6Prefix="<nil>"
level=info msg="Restoring endpoints..." subsys=daemon
level=info msg="Waiting until local node addressing before starting watchers depending on it" subsys=k8s-watcher
level=info msg="Endpoints restored" failed=0 restored=5 subsys=daemon
level=info msg="Addressing information:" subsys=daemon
level=info msg="  Cluster-Name: default" subsys=daemon
level=info msg="  Cluster-ID: 0" subsys=daemon
level=info msg="  Local node-name: node-adm-01" subsys=daemon
level=info msg="  Node-IPv6: <nil>" subsys=daemon
level=info msg="  External-Node IPv4: 159.223.122.251" subsys=daemon
level=info msg="  Internal-Node IPv4: 10.42.0.87" subsys=daemon
level=info msg="  IPv4 allocation prefix: 10.42.0.0/24" subsys=daemon
level=info msg="  IPv4 native routing prefix: 10.42.0.0/23" subsys=daemon
level=info msg="  Loopback IPv4: 169.254.42.1" subsys=daemon
level=info msg="  Local IPv4 addresses:" subsys=daemon
level=info msg="  - 159.223.122.251" subsys=daemon
level=info msg="  - 10.10.0.7" subsys=daemon
level=info msg="  - 10.116.0.4" subsys=daemon
level=info msg="  - 10.42.0.87" subsys=daemon
level=info msg="Adding local node to cluster" node="{node-adm-01 default [{InternalIP 159.223.122.251} {CiliumInternalIP 10.42.0.87}] 10.42.0.0/24 [] <nil> [] 10.42.0.244 <nil> <nil> <nil> 0 local 0 map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux egress.rke2.io/cluster:true kubernetes.io/arch:amd64 kubernetes.io/hostname:node-adm-01 kubernetes.io/os:linux node-role.kubernetes.io/control-plane:true node-role.kubernetes.io/etcd:true node-role.kubernetes.io/master:true] 1 }" subsys=nodediscovery
level=info msg="Creating or updating CiliumNode resource" node=node-adm-01 subsys=nodediscovery
level=info msg="Waiting until all pre-existing resources have been received" subsys=k8s-watcher
level=info msg="Initializing identity allocator" subsys=identity-cache
level=info msg="Allocating identities between range" cluster-id=0 max=65535 min=256 subsys=identity-cache
level=info msg="Cluster-ID is not specified, skipping ClusterMesh initialization" subsys=daemon
level=info msg="Setting up BPF datapath" bpfClockSource=ktime bpfInsnSet=v3 subsys=datapath-loader
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.core.bpf_jit_enable sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.all.rp_filter sysParamValue=0
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.fib_multipath_use_neigh sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=kernel.unprivileged_bpf_disabled sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=kernel.timer_migration sysParamValue=0
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_host.forwarding sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_host.rp_filter sysParamValue=0
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_host.accept_local sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_host.send_redirects sysParamValue=0
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_net.forwarding sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_net.rp_filter sysParamValue=0
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_net.accept_local sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_net.send_redirects sysParamValue=0
level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
level=info msg="Iptables rules installed" subsys=iptables
level=info msg="Adding new proxy port rules for cilium-dns-egress:41577" proxy port name=cilium-dns-egress subsys=proxy
level=info msg="Iptables proxy rules installed" subsys=iptables
level=info msg="Serving cilium node monitor v1.2 API at unix:///var/run/cilium/monitor1_2.sock" subsys=monitor-agent
level=info msg="Validating configured node address ranges" subsys=daemon
level=info msg="Starting connection tracking garbage collector" subsys=daemon
level=info msg="Starting IP identity watcher" subsys=ipcache
level=info msg="Initial scan of connection tracking completed" subsys=ct-gc
level=info msg="Regenerating restored endpoints" numRestored=5 subsys=daemon
level=info msg="New endpoint" containerID=fcda792a77 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=569 identity=26027 ipv4=10.42.0.83 ipv6= k8sPodName=kube-system/hubble-ui-64bd8dcbf5-zdj7c subsys=endpoint
level=info msg="New endpoint" containerID=65e91cb153 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=2710 identity=363 ipv4=10.42.0.84 ipv6= k8sPodName=kube-system/rke2-coredns-rke2-coredns-autoscaler-768bfc5985-zzxk6 subsys=endpoint
level=info msg="New endpoint" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=2163 identity=1 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="New endpoint" containerID=982ad4116a datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=1163 identity=19399 ipv4=10.42.0.173 ipv6= k8sPodName=kube-system/hubble-relay-59d9bc9bbf-lsr9g subsys=endpoint
level=info msg="New endpoint" containerID=f1db56daf5 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=823 identity=14223 ipv4=10.42.0.143 ipv6= k8sPodName=kube-system/rke2-coredns-rke2-coredns-58fd75f64b-w2jsp subsys=endpoint
level=info msg="Successfully restored endpoint. Scheduling regeneration" endpointID=823 subsys=daemon
level=info msg="Successfully restored endpoint. Scheduling regeneration" endpointID=1163 subsys=daemon
level=info msg="Successfully restored endpoint. Scheduling regeneration" endpointID=2163 subsys=daemon
level=info msg="Successfully restored endpoint. Scheduling regeneration" endpointID=2710 subsys=daemon
level=info msg="Successfully restored endpoint. Scheduling regeneration" endpointID=569 subsys=daemon
level=info msg="Removed endpoint" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=419 identity=4 ipv4=10.42.0.139 ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Launching Cilium health daemon" subsys=daemon
level=info msg="Conntrack garbage collector interval recalculated" deleteRatio=0.0008697509765625 newInterval=7m30s subsys=map-ct
level=info msg="Datapath signal listener running" subsys=signal
level=info msg="Reusing existing global key" key="k8s:app.kubernetes.io/instance=rke2-coredns;k8s:app.kubernetes.io/name=rke2-coredns;k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name=kube-system;k8s:io.cilium.k8s.policy.cluster=default;k8s:io.cilium.k8s.policy.serviceaccount=coredns;k8s:io.kubernetes.pod.namespace=kube-system;k8s:k8s-app=kube-dns;" subsys=allocator
level=info msg="Reusing existing global key" key="k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name=kube-system;k8s:io.cilium.k8s.policy.cluster=default;k8s:io.cilium.k8s.policy.serviceaccount=hubble-relay;k8s:io.kubernetes.pod.namespace=kube-system;k8s:k8s-app=hubble-relay;" subsys=allocator
level=info msg="Reusing existing global key" key="k8s:app.kubernetes.io/instance=rke2-coredns;k8s:app.kubernetes.io/name=rke2-coredns-autoscaler;k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name=kube-system;k8s:io.cilium.k8s.policy.cluster=default;k8s:io.cilium.k8s.policy.serviceaccount=rke2-coredns-rke2-coredns-autoscaler;k8s:io.kubernetes.pod.namespace=kube-system;k8s:k8s-app=kube-dns-autoscaler;" subsys=allocator
level=info msg="Reusing existing global key" key="k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name=kube-system;k8s:io.cilium.k8s.policy.cluster=default;k8s:io.cilium.k8s.policy.serviceaccount=hubble-ui;k8s:io.kubernetes.pod.namespace=kube-system;k8s:k8s-app=hubble-ui;" subsys=allocator
level=info msg="Launching Cilium health endpoint" subsys=daemon
level=info msg="Serving prometheus metrics on :9962" subsys=daemon
level=info msg="Started healthz status API server" address="127.0.0.1:9879" subsys=daemon
level=info msg="Initializing Cilium API" subsys=daemon
level=info msg="Successfully removed non-GC'ed neighbor entries previously installed by cilium-agent" count=1 subsys=linux-datapath
level=info msg="Daemon initialization completed" bootstrapTime=11.527895553s subsys=daemon
level=info msg="Serving cilium API at unix:///var/run/cilium/cilium.sock" subsys=daemon
level=info msg="Configuring Hubble server" eventQueueSize=2048 maxFlows=4095 subsys=hubble
level=info msg="Beginning to read perf buffer" startTime="2023-01-05 22:31:17.567378188 +0000 UTC m=+11.612284586" subsys=monitor-agent
level=info msg="Starting local Hubble server" address="unix:///var/run/cilium/hubble.sock" subsys=hubble
level=info msg="Starting Hubble server" address=":4244" subsys=hubble
level=info msg="Delete endpoint request" containerID=982ad4116a subsys=daemon
level=info msg="Delete endpoint request" containerID=fcda792a77 subsys=daemon
level=info msg="New endpoint" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=723 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Resolving identity labels (blocking)" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=723 identityLabels="reserved:health" ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Identity of endpoint changed" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=723 identity=4 identityLabels="reserved:health" ipv4= ipv6= k8sPodName=/ oldIdentity="no identity" subsys=endpoint
level=info msg="Compiled new BPF template" BPFCompilationTime=7.393956983s file-path=/var/run/cilium/state/templates/ba2e4c93bd72d7f982d7b0f1c92561593ebbb057302fde4381d1dd27e40de7c8/bpf_lxc.o subsys=datapath-loader
level=info msg="Rewrote endpoint BPF program" containerID=f1db56daf5 datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=823 identity=14223 ipv4=10.42.0.143 ipv6= k8sPodName=kube-system/rke2-coredns-rke2-coredns-58fd75f64b-w2jsp subsys=endpoint
level=info msg="Restored endpoint" endpointID=823 ipAddr="[10.42.0.143 ]" subsys=endpoint
level=info msg="Rewrote endpoint BPF program" containerID=982ad4116a datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=1163 identity=19399 ipv4=10.42.0.173 ipv6= k8sPodName=kube-system/hubble-relay-59d9bc9bbf-lsr9g subsys=endpoint
level=info msg="Restored endpoint" endpointID=1163 ipAddr="[10.42.0.173 ]" subsys=endpoint
level=info msg="Releasing key" key="[k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name=kube-system k8s:io.cilium.k8s.policy.cluster=default k8s:io.cilium.k8s.policy.serviceaccount=hubble-relay k8s:io.kubernetes.pod.namespace=kube-system k8s:k8s-app=hubble-relay]" subsys=allocator
level=info msg="Removed endpoint" containerID=982ad4116a datapathPolicyRevision=1 desiredPolicyRevision=1 endpointID=1163 identity=19399 ipv4=10.42.0.173 ipv6= k8sPodName=kube-system/hubble-relay-59d9bc9bbf-lsr9g subsys=endpoint
level=info msg="Rewrote endpoint BPF program" containerID=65e91cb153 datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=2710 identity=363 ipv4=10.42.0.84 ipv6= k8sPodName=kube-system/rke2-coredns-rke2-coredns-autoscaler-768bfc5985-zzxk6 subsys=endpoint
level=info msg="Restored endpoint" endpointID=2710 ipAddr="[10.42.0.84 ]" subsys=endpoint
level=info msg="Rewrote endpoint BPF program" containerID=fcda792a77 datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=569 identity=26027 ipv4=10.42.0.83 ipv6= k8sPodName=kube-system/hubble-ui-64bd8dcbf5-zdj7c subsys=endpoint
level=info msg="Restored endpoint" endpointID=569 ipAddr="[10.42.0.83 ]" subsys=endpoint
level=info msg="Releasing key" key="[k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name=kube-system k8s:io.cilium.k8s.policy.cluster=default k8s:io.cilium.k8s.policy.serviceaccount=hubble-ui k8s:io.kubernetes.pod.namespace=kube-system k8s:k8s-app=hubble-ui]" subsys=allocator
level=info msg="Removed endpoint" containerID=fcda792a77 datapathPolicyRevision=1 desiredPolicyRevision=1 endpointID=569 identity=26027 ipv4=10.42.0.83 ipv6= k8sPodName=kube-system/hubble-ui-64bd8dcbf5-zdj7c subsys=endpoint
level=info msg="Rewrote endpoint BPF program" containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=723 identity=4 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Compiled new BPF template" BPFCompilationTime=14.976655824s file-path=/var/run/cilium/state/templates/f91be9a0b7c3628d8d179a39113d847d2baee08d429f0aff443b9b8d8eb10917/bpf_host.o subsys=datapath-loader
level=info msg="Rewrote endpoint BPF program" containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=2163 identity=1 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Restored endpoint" endpointID=2163 ipAddr="[ ]" subsys=endpoint
level=info msg="Finished regenerating restored endpoints" regenerated=5 subsys=daemon total=5
level=info msg="Serving cilium health API at unix:///var/run/cilium/health.sock" subsys=health-server
level=info msg="Conntrack garbage collector interval recalculated" deleteRatio=0.0023193359375 newInterval=11m15s subsys=map-ct
```

### Anything else?

Output of `cilium-health status`:
```
root@node-adm-02:/home/cilium# cilium-health status
Probe time:   2023-01-05T22:32:53Z
Nodes:
  node-adm-02 (localhost):
    Host connectivity to 104.248.228.32:
      ICMP to stack:   OK, RTT=221.789µs
      HTTP to agent:   OK, RTT=456.703µs
    Endpoint connectivity to 10.42.1.74:
      ICMP to stack:   OK, RTT=225.074µs
      HTTP to agent:   OK, RTT=357.635µs
  node-adm-01:
    Host connectivity to 159.223.122.251:
      ICMP to stack:   OK, RTT=591.575µs
      HTTP to agent:   OK, RTT=1.184662ms
    Endpoint connectivity to 10.42.0.244:
      ICMP to stack:   Connection timed out
      HTTP to agent:   Get "http://10.42.0.244:4240/hello": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

Output of `cilium status --verbose`:
```
KVStore:                Ok   Disabled
Kubernetes:             Ok   1.24 (v1.24.9+rke2r1) [linux/amd64]
Kubernetes APIs:        ["cilium/v2::CiliumClusterwideNetworkPolicy", "cilium/v2::CiliumEndpoint", "cilium/v2::CiliumNetworkPolicy", "cilium/v2::CiliumNode", "core/v1::Namespace", "core/v1::Node", "core/v1::Pods", "core/v1::Service", "discovery/v1::EndpointSlice", "networking.k8s.io/v1::NetworkPolicy"]
KubeProxyReplacement:   Strict   [eth0 104.248.228.32 (Direct Routing), eth1 10.116.0.5, tun159223122251 104.248.228.32, tun178128144132 104.248.228.32]
Host firewall:          Disabled
CNI Chaining:           none
Cilium:                 Ok   1.12.5 (v1.12.5-701acde)
NodeMonitor:            Listening for events on 2 CPUs with 64x4096 of shared memory
Cilium health daemon:   Ok   
IPAM:                   IPv4: 7/254 allocated from 10.42.1.0/24, 
Allocated addresses:
  10.42.1.1 (kube-system/hubble-relay-59d9bc9bbf-7qm76)
  10.42.1.111 (default/nginx[restored])
  10.42.1.155 (router)
  10.42.1.161 (kube-system/rke2-metrics-server-74f878b999-mtsks[restored])
  10.42.1.210 (kube-system/hubble-ui-64bd8dcbf5-rkcvp)
  10.42.1.74 (health)
  10.42.1.89 (kube-system/rke2-coredns-rke2-coredns-58fd75f64b-xkcz5[restored])
BandwidthManager:       Disabled
Host Routing:           BPF
Masquerading:           BPF   [eth0, eth1, tun159223122251, tun178128144132]   10.42.0.0/23 [IPv4: Enabled, IPv6: Disabled]
Clock Source for BPF:   ktime
Controller Status:      43/43 healthy
  Name                                          Last success   Last error   Count   Message
  bpf-map-sync-cilium_lxc                       5s ago         never        0       no error   
  cilium-health-ep                              54s ago        never        0       no error   
  dns-garbage-collector-job                     5s ago         never        0       no error   
  endpoint-226-regeneration-recovery            never          never        0       no error   
  endpoint-2378-regeneration-recovery           never          never        0       no error   
  endpoint-2515-regeneration-recovery           never          never        0       no error   
  endpoint-2964-regeneration-recovery           never          never        0       no error   
  endpoint-490-regeneration-recovery            never          never        0       no error   
  endpoint-52-regeneration-recovery             never          never        0       no error   
  endpoint-676-regeneration-recovery            never          never        0       no error   
  endpoint-gc                                   2m6s ago       never        0       no error   
  ipcache-inject-labels                         2m1s ago       2m4s ago     0       no error   
  k8s-heartbeat                                 6s ago         never        0       no error   
  link-cache                                    10s ago        never        0       no error   
  metricsmap-bpf-prom-sync                      0s ago         never        0       no error   
  neighbor-table-refresh                        25s ago        never        0       no error   
  resolve-identity-226                          1m54s ago      never        0       no error   
  resolve-identity-2964                         1m54s ago      never        0       no error   
  resolve-identity-490                          1m54s ago      never        0       no error   
  restoring-ep-identity (2378)                  1m55s ago      never        0       no error   
  restoring-ep-identity (2515)                  1m55s ago      never        0       no error   
  restoring-ep-identity (52)                    1m55s ago      never        0       no error   
  restoring-ep-identity (676)                   1m55s ago      never        0       no error   
  sync-endpoints-and-host-ips                   55s ago        never        0       no error   
  sync-lb-maps-with-k8s-services                1m55s ago      never        0       no error   
  sync-policymap-226                            48s ago        never        0       no error   
  sync-policymap-2378                           48s ago        never        0       no error   
  sync-policymap-2515                           48s ago        never        0       no error   
  sync-policymap-2964                           48s ago        never        0       no error   
  sync-policymap-490                            47s ago        never        0       no error   
  sync-policymap-52                             48s ago        never        0       no error   
  sync-policymap-676                            28s ago        never        0       no error   
  sync-to-k8s-ciliumendpoint (226)              4s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (2378)             5s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (2515)             5s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (2964)             4s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (490)              4s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (52)               5s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (676)              5s ago         never        0       no error   
  template-dir-watcher                          never          never        0       no error   
  waiting-initial-global-identities-ep (2378)   1m55s ago      never        0       no error   
  waiting-initial-global-identities-ep (2515)   1m55s ago      never        0       no error   
  waiting-initial-global-identities-ep (52)     1m55s ago      never        0       no error   
Proxy Status:            OK, ip 10.42.1.155, 0 redirects active on ports 10000-20000
Global Identity Range:   min 256, max 65535
Hubble:                  Ok   Current/Max Flows: 1668/4095 (40.73%), Flows/s: 14.31   Metrics: Disabled
KubeProxyReplacement Details:
  Status:                 Strict
  Socket LB:              Enabled
  Socket LB Protocols:    TCP, UDP
  Devices:                eth0 104.248.228.32 (Direct Routing), eth1 10.116.0.5, tun159223122251 104.248.228.32, tun178128144132 104.248.228.32
  Mode:                   DSR
  Backend Selection:      Random
  Session Affinity:       Enabled
  Graceful Termination:   Enabled
  NAT46/64 Support:       Disabled
  XDP Acceleration:       Disabled
  Services:
  - ClusterIP:      Enabled
  - NodePort:       Enabled (Range: 30000-32767) 
  - LoadBalancer:   Enabled 
  - externalIPs:    Enabled 
  - HostPort:       Enabled
BPF Maps:   dynamic sizing: on (ratio: 0.002500)
  Name                          Size
  Non-TCP connection tracking   65536
  TCP connection tracking       131072
  Endpoint policy               65535
  Events                        2
  IP cache                      512000
  IP masquerading agent         16384
  IPv4 fragmentation            8192
  IPv4 service                  65536
  IPv6 service                  65536
  IPv4 service backend          65536
  IPv6 service backend          65536
  IPv4 service reverse NAT      65536
  IPv6 service reverse NAT      65536
  Metrics                       1024
  NAT                           131072
  Neighbor table                131072
  Global policy                 16384
  Per endpoint policy           65536
  Session affinity              65536
  Signal                        2
  Sockmap                       65535
  Sock reverse NAT              65536
  Tunnel                        65536
Encryption:                 Disabled
Cluster health:             2/2 reachable     (2023-01-05T22:31:19Z)
  Name                      IP                Node        Endpoints
  node-adm-02 (localhost)   104.248.228.32    reachable   reachable
  node-adm-01               159.223.122.251   reachable   reachable
  ```

Output of `ip route` on node 1:
```
root@node-adm-01:~# ip route
default via 159.223.112.1 dev eth0 onlink
10.10.0.0/16 dev eth0 proto kernel scope link src 10.10.0.7
10.42.0.0/24 via 10.42.0.87 dev cilium_host src 10.42.0.87
10.42.0.87 dev cilium_host scope link
10.42.1.0/24 dev tun-10424822832 proto 17 src 159.223.122.251
10.116.0.0/20 dev eth1 proto kernel scope link src 10.116.0.4
159.223.112.0/20 dev eth0 proto kernel scope link src 159.223.122.251
```

Output of `ip route` on node 2:
```
root@node-adm-02:~# ip route
default via 104.248.224.1 dev eth0 onlink
10.10.0.0/16 dev eth0 proto kernel scope link src 10.10.0.8
10.20.0.0/24 dev tun178128144132 proto 17 src 104.248.228.32
10.42.0.0/24 dev tun159223122251 proto 17 src 104.248.228.32
10.42.1.0/24 via 10.42.1.155 dev cilium_host src 10.42.1.155
10.42.1.155 dev cilium_host scope link
10.116.0.0/20 dev eth1 proto kernel scope link src 10.116.0.5
104.248.224.0/20 dev eth0 proto kernel scope link src 104.248.228.32
```

Cilium Configuration:
```
helm install cilium cilium/cilium --version 1.12.5 \                                                     
--namespace kube-system \
--set operator.replicas=1 \
--set ipam.mode=kubernetes \
--set tunnel=disabled \
--set bpf.masquerade=true \
--set ipv4NativeRoutingCIDR=10.42.0.0/16 \
--set hubble.enabled=true \
--set hubble.relay.enabled=true \
--set hubble.ui.enabled=true \
--set kubeProxyReplacement=strict \
--set loadBalancer.mode=dsr \
--set k8sServiceHost=159.223.122.251 \
--set k8sServicePort=6443
```

Kube-router Configuration:
```yaml
apiVersion: apps/v1
kind: DaemonSet
metadata:
  labels:
    k8s-app: kube-router
    tier: node
  name: kube-router
  namespace: kube-system
spec:
  selector:
    matchLabels:
      k8s-app: kube-router
      tier: node
  template:
    metadata:
      labels:
        k8s-app: kube-router
        tier: node
    spec:
      priorityClassName: system-node-critical
      serviceAccountName: kube-router
      containers:
      - name: kube-router
        image: docker.io/cloudnativelabs/kube-router
        imagePullPolicy: Always
        args:
          - "--run-router=true"
          - "--run-firewall=false"
          - "--run-service-proxy=false"
          - "--enable-cni=false"
          - "--enable-pod-egress=false"
          - "--enable-ibgp=true"
          - "--enable-overlay=true"
          - "--advertise-cluster-ip=true"
          - "--advertise-external-ip=true"
          - "--advertise-loadbalancer-ip=true"
        env:
        - name: NODE_NAME
          valueFrom:
            fieldRef:
              fieldPath: spec.nodeName
        livenessProbe:
          httpGet:
            path: /healthz
            port: 20244
          initialDelaySeconds: 10
          periodSeconds: 3
        resources:
          requests:
            cpu: 250m
            memory: 250Mi
        securityContext:
          privileged: true
        volumeMounts:
        - name: xtables-lock
          mountPath: /run/xtables.lock
          readOnly: false
      hostNetwork: true
      tolerations:
      - effect: NoSchedule
        operator: Exists
      - key: CriticalAddonsOnly
        operator: Exists
      - effect: NoExecute
        operator: Exists
      volumes:
        - name: xtables-lock
          hostPath:
            path: /run/xtables.lock
            type: FileOrCreate
```

Rke Server Configuration:
```
cni: "none"
node-ip: "10.116.0.4"
node-external-ip: "159.223.122.251"
disable-kube-proxy: true 
disable: "rke2-ingress-nginx"
disable-cloud-controller: true
```

Rke Agent Configuration:
```
server: "https://10.116.0.4:9345"
node-ip: "10.116.0.5"
node-external-ip: "104.248.228.32"
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
